app/utils/app_info.py

```diff
@@ -13,36 +16,34 @@ class AppInfo:
     using the `platformdirs` package, ensuring platform-specific conventions are adhered to.
 
     Examples:
-        >>> app_info = AppInfo(__file__)
         >>> print(app_info.app_name)
         >>> print(app_info.app_storage_folder)
     """
 
     _instance: Optional["AppInfo"] = None
 
-    def __new__(cls, main_file: Optional[str] = None) -> "AppInfo":
+    def __new__(cls) -> "AppInfo":
         """
         Create a new instance or return the existing singleton instance of the `AppInfo` class.
         """
         if not cls._instance:
             cls._instance = super(AppInfo, cls).__new__(cls)
         return cls._instance
 
-    def __init__(self, main_file: Optional[str] = None) -> None:
+    def __init__(self) -> None:
         """
         Initialize the `AppInfo` instance, setting application metadata and determining important directories.
 
-        Args:
-            main_file (Optional[str]): Path to the main application file (i.e., __file__ from __main__).
-
         Raises:
-            ValueError: If `main_file` is not provided during the first initialization.
+            Exception: If the main file path cannot be determined.
         """
         if hasattr(self, "_is_initialized") and self._is_initialized:
             return
 
+        main_file = sys.modules["__main__"].__file__
```

Comment:
No changes needed: there are cases where `__main__` does not exist (e.g. interactive interpreter or embedded Python environment). Wrapping this with a try/except for `KeyError, AttributeError` would make safer

Reply:
So main here refers to the literal file that is called "__main__" that acts as the entry point for the entire app, not just this specific file. If that module doesn't exist, then the entire app isn't goimg to work anyways. I tested in interpreted mode albeit with the normal app entry point and it seems fine.